subsys/caf/modules/sensor_sampler.c

```diff
@@ -19,11 +20,29 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_CAF_SENSOR_SAMPLER_LOG_LEVEL);
 
 #define SAMPLE_THREAD_STACK_SIZE	CONFIG_CAF_SENSOR_SAMPLER_THREAD_STACK_SIZE
 #define SAMPLE_THREAD_PRIORITY		CONFIG_CAF_SENSOR_SAMPLER_THREAD_PRIORITY
+#define SAMPLE_BUF_SIZE			CONFIG_CAF_SENSOR_SAMPLER_DATA_BUF_SIZE
+
+static struct k_sem sample;
+static struct sensor_value buf[SAMPLE_BUF_SIZE];
+static struct sensor_value *next_buf = buf;
```

Comment:
I am against pointer arithmetics. It's easier to make a mistake using it and mistake can lead to undefined behavior.

Reply:
:(

Reply:
Sorry, I truly am. It is really much simpler to work with numbers plus if you use unsigned int you won't get UB on arithmetic ops so the only thing left to worry about is access beyond array bounds.